src/i2p.cpp

```diff
@@ -355,29 +366,47 @@ void Session::CreateIfNotCreatedAlready()
         return;
     }
 
-    Log("Creating SAM session with %s", m_control_host.ToString());
+    const auto session_type = m_transient ? "transient" : "persistent";
+    const auto session_id = GetRandHash().GetHex().substr(0, 10); // full is an overkill, too verbose in the logs
```

Comment:
```suggestion
    const auto session_id{GetRandHash().GetHex().substr(0, 10)}; // a full id would be overly verbose in the logs
```

(not sure if braced initialization is better here)

(in any case: `s/an overkill/overkill/`)

Reply:
`{}` forces stricter type checks. E.g. `Foo x{expression_that_returns_Bar};` but it is pointless when `auto` is used. I find it more readable to use `=`.

Reply:
maybe s/const auto/const auto&/ (I'm not sure if compilers curerntly detect and optimize this to avoid a copy)